doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md

```diff
@@ -6,5 +6,131 @@ nav: /:path/Programming-guide/Key-platform-features/Devices-provisioning-and-reg
+
+When Tenant Admin create application he can specify credential service for this application. By default used "Trustful".
+
+## Creating custom credentials service
+
+1. First of all you should create class which implements all method of CredentialsService interface
+
+```java
+
+package org.myproject;
+
+@Service
+public class CustomCredentialsService implements CredentialsService {
+
+    /**
+     * Provide credentials information to the external system.
+     */
+    @Override
+    public CredentialsDto provideCredentials(String applicationId, CredentialsDto credentials) throws CredentialsServiceException {
+        // implementation
+    }
+
+    /**
+     * Returns the credentials by ID.
+     */
+    @Override
+    public Optional<CredentialsDto> lookupCredentials(String applicationId, String credentialsId) throws CredentialsServiceException {
+        // implementation
+    }
+
+    /**
+     * Sets the status of the given credentials to CredentialsStatus.IN_USE
+     */
+    @Override
+    public void markCredentialsInUse(String applicationId, String credentialsId) throws CredentialsServiceException {
+        // implementation
+    }
+
+    /**
+     * Revokes the given credentials by setting their status to CredentialsStatus.REVOKED
+     */
+    @Override
+    public void markCredentialsRevoked(String applicationId, String credentialsId) throws CredentialsServiceException {
+        // implementation
+    }
+}
+
+```
+
+Exist three credential status
+
+* AVAILABLE
+* IN_USE
+* REVOKED
+
+2. In in /usr/lib/kaa-node/conf/kaaNodeContext.xml register CredentialsServiceLocator for your new credential service 
+
+```xml
+
+<bean id="customCredentialsServiceLocator" class="org.kaaproject.kaa.server.node.service.credentials.InternalCredentialsServiceLocator">
+    <constructor-arg ref="org.myproject.customCredentialsService"/>
+</bean>
+    
+<util:map id="credentialsServiceLocatorMap">
+    <entry key="Trustful" value-ref="trustfulCredentialsServiceLocator"/>
+    <entry key="Internal" value-ref="internalCredentialsServiceLocator"/>
+    <entry key="NEW" value-ref="customCredentialsServiceLocator"/>                          
+</util:map>
+
+```
+
+Value of key which was added you will see in Admin UI.
+
+![credential](credential.png)
+
+3. In in /usr/lib/kaa-node/conf/common-dao-context.xml
+
+```xml
+
+<bean id="customCredentialsService" class="org.myproject.CustomCredentialsService"/>
+<alias name="customCredentialsService" alias="NEW Credentials Service"/>
+
+```
+
+##  Custom credentials service provisioning
+
+To provision your credentials service, do the following:
+
+Build your application using next command: 
+
+```bash
+
+$ mvn clean install
+
+```
+
+Place your application *.jar from /target folder into the /usr/lib/kaa-node/lib folder.
+Specify additional package to scan kaa plugins configuration in parameter additional_plugins_scan_package.
```

Comment:
please specify some command how to edit configuration file ( no name of configuration file mentioned here)